core/src/main/java/io/questdb/griffin/engine/table/FilterOnExcludedValuesRecordCursorFactory.java

```diff
@@ -80,20 +91,32 @@ public FilterOnExcludedValuesRecordCursorFactory(
         this.filter = filter;
         cursorFactoriesIdx = new int[]{0};
         cursorFactories = new ObjList<>(nKeyValues);
-        if (orderByMnemonic == OrderByMnemonic.ORDER_BY_INVARIANT) {
+        if (orderByMnemonic == OrderByMnemonic.ORDER_BY_INVARIANT && !orderByTimestamp) {
             cursor = new DataFrameRecordCursorImpl(new SequentialRowCursorFactory(cursorFactories, cursorFactoriesIdx), false, filter, columnIndexes);
         } else {
             cursor = new DataFrameRecordCursorImpl(new HeapRowCursorFactory(cursorFactories, cursorFactoriesIdx), false, filter, columnIndexes);
```

Comment:
What if it the order is by timestamp descending here, will `HeapRowCursorFactory` do descending order?

Reply:
Conditionals in SqlCodeGenerator should prevent that from happening . HeapRowCursorFactory doesn't support descending order . I can add it if needed .

Reply:
If it doesn't support desc order, we should use `SequentialRowCursorFactory` when the order is required to be descending, it's cheaper.

Reply:
That depends on whether data is in memory or disk . If it's the latter then heap cursor should generate more sequential IO pattern and could be faster.

Reply:
Why don't we use it always then?